packages/shared/src/hooks/useAutoExpand.ts

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+
+const useAutoExpand = () => {
+  const ref = React.useRef(null);
+  const [height, setHeight] = React.useState(0);
+
+  const updateHeight = () => {
+    if (ref.current) {
+      const windowHeight = window.innerHeight;
+      const divTopOffset = ref.current.getBoundingClientRect().top;
+      const availableHeight = windowHeight - divTopOffset;
+      setHeight(availableHeight);
```

Comment:
just a nit (can be skipped as well):
```suggestion
      availableHeight !== height && setHeight(availableHeight);
```

Reply:
This should only be triggered when the window size is changed so the guard might not be necessary.